src/components/Meme.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+
+const Meme = () => {
+  const [meme, setMeme] = React.useState({
+    topText: "",
+    bottomText: "",
+    randomImage: "http://i.imgflip.com/1bij.jpg",
+  });
+
+  const [allMemes, setAllMemes] = useState([]);
+  useEffect(() => {
+    async function getMemes() {
+      const res = await fetch("https://api.imgflip.com/get_memes");
+      const data = await res.json();
+      setAllMemes(data.data.memes);
+    }
+    getMemes();
+  }, []);
+
+  const getMemeImage = () => {
+    const randomNumber = Math.floor(Math.random() * allMemes.length);
+    const url = allMemes[randomNumber].url;
```

Comment:
Improve this by using the extraction instead of accessing it through multiple levels of nesting.
``` suggestion
    const {url} = allMemes[randomNumber];
```